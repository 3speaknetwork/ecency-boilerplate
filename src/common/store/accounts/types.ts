import { ExtendedAccount } from "@esteemapp/dhive";

export interface Account extends ExtendedAccount {}

export interface State {
  data: Account;
  loading: boolean;
  error: boolean;
}

export enum ActionTypes {
  FETCH = "@account/FETCH",
  FETCHED = "@account/FETCHED",
  FETCH_ERROR = "@account/FETCH_ERROR"
}

export interface FetchAction {
  type: ActionTypes.FETCH;
}

export interface FetchedAction {
  type: ActionTypes.FETCHED;
  data: Account;
}

export interface FetchErrorAction {
  type: ActionTypes.FETCH_ERROR;
}

export type Actions = FetchAction | FetchedAction | FetchErrorAction;