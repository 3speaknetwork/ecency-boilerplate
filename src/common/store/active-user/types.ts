import { Account } from "../accounts/types";

export interface ActiveUser {
  name: string;
  data: Account;
}

export enum ActionTypes {
  RELOAD = "@active-user/RELOAD",
  UPDATE = "@active-user/UPDATE",
}

export interface ReloadAction {
  type: ActionTypes.RELOAD;
}

export interface UpdateAction {
  type: ActionTypes.UPDATE;
  data: Account;
}

export type Actions = ReloadAction | UpdateAction;
