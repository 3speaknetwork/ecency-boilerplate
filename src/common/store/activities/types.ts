interface BaseActivity {
    event: "notify";
    source: string;
    target: string;
    timestamp: string;
    extra: any;
}

export interface VoteActivity extends BaseActivity {
    type: "vote",
    extra: {
        permlink: string;
        weight: number;
        title: string | null;
        img_url: string | null;
    }
}

export interface MentionActivity extends BaseActivity {
    type: "mention",
    extra: {
        permlink: string;
        is_post: 0 | 1;
        title: string | null;
        img_url: string | null;
    }
}

export interface FollowActivity extends BaseActivity {
    type: "follow",
    extra: {
        what: string[]
    }
}

export interface ReplyActivity extends BaseActivity {
    type: "reply",
    extra: {
        title: string;
        body: string;
        json_metadata: string;
        permlink: string;
        parent_author: string;
        parent_permlink: string;
        parent_title: string | null;
        parent_img_url: string | null;
    }
}

export interface ReblogActivity extends BaseActivity {
    type: "reblog";
    extra: {
        permlink: string;
        title: string | null;
        img_url: string | null;
    }

}

export interface TransferActivity extends BaseActivity {
    type: "transfer";
    extra: {
        amount: string;
        memo: string;
    }
}


export type Activity = VoteActivity | MentionActivity | FollowActivity | ReplyActivity | ReblogActivity | TransferActivity;
