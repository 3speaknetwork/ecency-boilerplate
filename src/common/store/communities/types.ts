export interface Community {
    about: string,
    avatar_url: string,
    created_at: string,
    description: string,
    flag_text: string,
    id: number,
    is_nsfw: boolean,
    lang: string,
    name: string,
    num_authors: number,
    num_pending: number,
    subscribers: number,
    sum_pending: number,
    team: Array<Array<string>>,
    title: string,
    type_id: number
}

export interface State {
    list: Record<string, Community>,
}

export enum ActionTypes {
    FETCHED = '@communities/FETCHED',
}


export interface FetchedAction {
    type: ActionTypes.FETCHED;
    name: string,
    data: Community
}

export type Actions = FetchedAction; // | .. | ..
