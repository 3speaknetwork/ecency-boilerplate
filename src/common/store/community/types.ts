export interface Community {
  about: string;
  avatar_url: string;
  created_at: string;
  description: string;
  flag_text: string;
  id: number;
  is_nsfw: boolean;
  lang: string;
  name: string;
  num_authors: number;
  num_pending: number;
  subscribers: number;
  sum_pending: number;
  settings?: any;
  team: Array<Array<string>>;
  title: string;
  type_id: number;
}

export interface State extends Community {}

export enum ActionTypes {
  FETCHED = "@communities/FETCHED",
  RESET = "@communities/RESET",
}

export interface FetchedAction {
  type: ActionTypes.FETCHED;
  data: Community;
}

export interface ResetAction {
  type: ActionTypes.RESET;
}

export type Actions = FetchedAction | ResetAction;
