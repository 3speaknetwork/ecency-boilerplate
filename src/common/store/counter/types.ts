export interface State {
    val: number
}

export enum ActionTypes {
    SET_COUNTER = '@counter/SET_COUNTER',
    INCREMENT_COUNTER = '@counter/INCREMENT_COUNTER',
    DECREMENT_COUNTER = '@counter/DECREMENT_COUNTER',
}

export interface SetAction {
    type: ActionTypes.SET_COUNTER;
    value: string;
}

export interface IncrementAction {
    type: ActionTypes.INCREMENT_COUNTER;
}

export interface DecrementAction {
    type: ActionTypes.DECREMENT_COUNTER;
}

export type Actions = SetAction | IncrementAction | DecrementAction
