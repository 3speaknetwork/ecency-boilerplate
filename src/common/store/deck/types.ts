export interface DeckModel {
  data?: any;
  dataParams: any[];
  header: {
    title: string;
    icon: JSX.Element | null;
    reloading: boolean;
  };
  listItemComponent: any;
  createdAt: Date;
  // TODO: Add owner
}

export interface IdentifiableDeckModel extends DeckModel {
  id: string;
  content: string;
}

export enum ActionTypes {
  CREATE = "@deck/CREATE",
  SET_DATA =  "@deck/FETCH_DATA",
  SET_RELOADING = "@deck/SET_RELOADING",
  REORDER = "@deck/REORDER"
}

export interface CreateAction {
  type: ActionTypes.CREATE;
  data: [
    DeckModel['listItemComponent'],
    DeckModel['header']['title'],
    DeckModel['header']['icon'],
    DeckModel['dataParams']
  ];
}

export interface SetDataAction {
  type: ActionTypes.SET_DATA;
  data: {
    title: string;
    data: any;
  };
}

export interface SetReloadingAction {
  type: ActionTypes.SET_RELOADING;
  data: {
    title: string;
    isReloading: boolean;
  };
}

export interface ReOrderAction {
  type: ActionTypes.REORDER;
  data: {
    startIndex: number;
    endIndex: number;
  }
}

export type DeckState = {
  items: IdentifiableDeckModel[];
  isContentLoading: boolean;
};
export type Actions = CreateAction | SetDataAction | SetReloadingAction;