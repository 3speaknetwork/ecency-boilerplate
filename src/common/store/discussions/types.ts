import {Entry} from "../entries/types";

export interface Discussions {
    list: Record<string, Entry>;
    loading: boolean;
    error: boolean;
}

export enum ActionTypes {
    FETCH = "@discussions/FETCH",
    FETCHED = "@discussions/FETCHED",
    FETCH_ERROR = "@discussions/FETCH_ERROR",
    RESET = "@discussions/RESET",
}

export interface FetchAction {
    type: ActionTypes.FETCH;
}

export interface FetchedAction {
    type: ActionTypes.FETCHED;
    list: Record<string, Entry>;
}

export interface FetchErrorAction {
    type: ActionTypes.FETCH_ERROR;
}

export interface ResetAction {
    type: ActionTypes.RESET;
}

export type Actions = FetchAction | FetchedAction | FetchErrorAction | ResetAction;
