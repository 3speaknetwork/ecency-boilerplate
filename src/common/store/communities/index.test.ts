import reducer, {initialState, fetchedAct} from './index';

import {State} from './types';

let state: State = initialState;

it('1- default state', () => {
    expect(state).toMatchSnapshot();
});

it('1- fetched', () => {
    // @ts-ignore
    state = reducer(state, fetchedAct('hive-1234', {title: 'bitcoin'}));
    expect(state).toMatchSnapshot();
});

it('2- fetched', () => {
    // @ts-ignore
    state = reducer(state, fetchedAct('hive-2345', {title: 'photography'}));
    expect(state).toMatchSnapshot();
});
