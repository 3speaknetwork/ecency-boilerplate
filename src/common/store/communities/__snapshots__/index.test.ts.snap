// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default state 1`] = `
Object {
  "list": Object {},
}
`;

exports[`1- fetched 1`] = `
Object {
  "list": Object {
    "hive-1234": Object {
      "title": "bitcoin",
    },
  },
}
`;

exports[`2- fetched 1`] = `
Object {
  "list": Object {
    "hive-1234": Object {
      "title": "bitcoin",
    },
    "hive-2345": Object {
      "title": "photography",
    },
  },
}
`;
